src/texture/mod.rs

```diff
@@ -358,6 +358,38 @@ pub struct RawImage3d<'a, T: Clone + 'a> {
     pub format: ClientFormat,
 }
 
+impl<'a, T: Clone + 'a> RawImage3d<'a, T> {
+    ///Transforms a Vec<RawImage2d> into a RawImage3d
+    pub fn from_vec_raw2d(arr: &Vec<RawImage2d<'a, T>>) -> Result<RawImage3d<'a, T>, TextureMaybeSupportedCreationError> {
+        let depth   = arr.len();
+        let width   = arr[0].width;
+        let height  = arr[0].height;
+        let format  = arr[0].format;
+        let raw_data = {
+            let mut vec = Vec::<T>::with_capacity((width * height * depth as u32) as usize);
+            for i in arr {
+                if width != i.width || height != i.height {
+                    return Err(TextureMaybeSupportedCreationError::CreationError(TextureCreationError::DimensionsNotSupported));
```

Comment:
Ah, okay.  The exact errors aren't something I knew the reasons of (only wrote this because I was converting raw opengl calls to glium, and noticed this wasn't supported)

I hadn't realised you wanted the user to be the one to enforce consistent dimensions, not glium.  Makes sense though.